### PR TITLE
added cluster storage cypress test

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: https://issues.redhat.com/browse/RHOAIENG-3809

## Description
added cypress test for cluster storage page

## How Has This Been Tested?
npm run test

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/__tests__/cypress/cypress/e2e/projects/clusterStorage.cy.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+      expect(interception.request.url).not.to.include('?dryRun=All');
+    });
+
+    cy.get('@addClusterStorage.all').then((interceptions) => {
+      expect(interceptions).to.have.length(2);
+    });
+  });
+
+  it('Renders the page with cluster storage table data', () => {
+    initInterceptors({});
```
Comment (ppadti) [inline frontend/src/__tests__/cypress/cypress/e2e/projects/clusterStorage.cy.ts:133]: Could you please add sorting based on Name here

File: frontend/src/__tests__/cypress/cypress/e2e/projects/clusterStorage.cy.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+      mockPVCK8sResource({}),
+    ).as('editClusterStorage');
+
+    updateClusterStorageModal.findSubmitButton().click();
+    cy.wait('@editClusterStorage').then((interception) => {
+      expect(interception.request.url).to.include('?dryRun=All');
+      expect(interception.request.body).to.eql({
+        kind: 'PersistentVolumeClaim',
+        apiVersion: 'v1',
+        metadata: {
```
Comment (ppadti) [inline frontend/src/__tests__/cypress/cypress/e2e/projects/clusterStorage.cy.ts:168]: May be we could use `chai-subset` here after this [PR](https://github.com/opendatahub-io/odh-dashboard/pull/2643) gets merged.

**Review by ppadti — CHANGES_REQUESTED** [review]

**Comment by pnaik1** [comment]
Updated the PR

File: frontend/src/__tests__/cypress/cypress/pages/clusterStorage.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    this.find().find(`[data-label=Type]`).contains(name).should('exist');
+    return this;
+  }
+
+  findConnectedWorkbenches() {
+    return this.find().find('[data-label="Connected workbenches"]');
+  }
+
+  toggleExpandByIndex(index: number) {
+    this.find().find('button').should('have.attr', 'aria-label', 'Details').eq(index).click();
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/pages/clusterStorage.ts:15]: I believe you can use `findByRole` and `name` for this selector. Also, why do you need to check the `index` ? Shouldn't there only be one per row?
Ideally we use test
```suggestion
    this.find().findByRole('button', { name: 'Details' }).click();
```

File: frontend/src/__tests__/cypress/cypress/e2e/projects/ClusterStorage.cy.ts [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+      pathname: '/api/k8s/apis/kubeflow.org/v1/namespaces/test-project/notebooks',
+    },
+    mockK8sResourceList([mockNotebookK8sResource({})]),
+  );
+  cy.intercept(
+    {
+      pathname: '/api/namespaces/test-project/*',
+    },
+    { statusCode: 200, body: { applied: true } },
+  );
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/e2e/projects/ClusterStorage.cy.ts:72]: Why is this needed?

**Pull request merged**